### ISSUE TITLE
[Bug]: Inconstent highlighting on mobile

### ISSUE THREAD

**Issue by 17hoehbr** [mozilla/pdf.js#18931]
### Attach (recommended) or Link to PDF file

https://mozilla.github.io/pdf.js/web/viewer.html

### Web browser and its version

Chromite 128

### Operating system and its version

Android 14

### PDF.js version

4.7.76

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

1. Toggle the highlight tool on a mobile device (possibly touch devices in general)
2. While using the highlighter, make initial touch over the text layer and swipe down

https://github.com/user-attachments/assets/e27ba88a-5085-4add-8d0e-d9cda41eb8fb

### What is the expected behavior?

I expected the highlighter to do free highlighting, but it seems to conflict with the text selection. On desktop browsers, making this same gesture with the mouse immediately switches to text selection mode. 

[Screencast_20241021_022308.webm](https://github.com/user-attachments/assets/965130ea-0972-45ae-ae05-a77086e5d053)

### What went wrong?

On mobile devices, toggling text selection requires pressing and holding for a second before triggering. Therefore, nothing happens if you highlight over text. 

### Additional context

The most straight forward solution in my opinion would be to separate the text selection highlighting and the free-handed highlighting into two separate tools.

**Comment by mintxdp** [comment]
Yes, i have this also as there is very inconsistent highlighting and random jiggaly patterend highlight appears on certain movements of hands with highlight mode on.
